### PR TITLE
Add Fracture analysis : contact debonding example

### PR DESCRIPTION
**Pull request by dipinknair**
steps
- [x] draft the example
- [x] upload examples files
- [x] add assertion
- [x] add sections and description
- [x] add messages inside
- [x] add results and gif

### CODE REVIEW

File: examples/basic/fracture_analysis_contact_debonding.py [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+)
+
+ExtAPI.Graphics.Camera.SetFit()
+ExtAPI.Graphics.ExportImage(
+    os.path.join(cwd, "geometry.png"), image_export_format, settings_720p
+)
+display_image("geometry.png")
+
+# %%
+# Material import , named selections, materials and connections
```
Comment (klmcadams) [inline examples/basic/fracture_analysis_contact_debonding.py:92]:
```suggestion
# Material import, named selections, materials and connections
```

File: examples/basic/fracture_analysis_contact_debonding.py [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+# Define static structural analysis and settings
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+MODEL.AddStaticStructuralAnalysis()
+STATIC_STRUCTURAL = ExtAPI.DataModel.AnalysisByName("Static Structural")
+ANALYSIS_SETTINGS = STATIC_STRUCTURAL.AnalysisSettings
+SOLUTION = STATIC_STRUCTURAL.Solution
+MESH = Model.Mesh
+
+# Set Unit System
```
Comment (klmcadams) [inline examples/basic/fracture_analysis_contact_debonding.py:207]:
```suggestion
# Set unit system
```

File: examples/basic/fracture_analysis_contact_debonding.py [hunk 3]
```diff
@@ -0,0 +1,405 @@
...
+
+ExtAPI.Application.ActiveUnitSystem = MechanicalUnitSystem.StandardNMM
+
+# Set 2D Behavior
```
Comment (klmcadams) [inline examples/basic/fracture_analysis_contact_debonding.py:211]:
```suggestion
# Set 2D behavior
```

File: examples/basic/fracture_analysis_contact_debonding.py [hunk 4]
```diff
@@ -0,0 +1,405 @@
...
+
+GEOMETRY.Activate()
+GEOMETRY.Model2DBehavior = Model2DBehavior.PlaneStrain
+
+# Assign Material
```
Comment (klmcadams) [inline examples/basic/fracture_analysis_contact_debonding.py:216]:
```suggestion
# Assign material
```

File: examples/basic/fracture_analysis_contact_debonding.py [hunk 5]
```diff
@@ -0,0 +1,405 @@
...
+
+PART.Activate()
+PART.Material = MAT_BODY.Name
+
+# Define Contact Region
```
Comment (klmcadams) [inline examples/basic/fracture_analysis_contact_debonding.py:221]:
```suggestion
# Define contact region
```

File: examples/basic/fracture_analysis_contact_debonding.py [hunk 6]
```diff
@@ -0,0 +1,405 @@
...
+MESH.GenerateMesh()
+
+ExtAPI.Graphics.Camera.SetFit()
+ExtAPI.Graphics.ExportImage(
+    os.path.join(cwd, "mesh.png"), image_export_format, settings_720p
+)
+display_image("mesh.png")
+
+# %%
+# Add Contact Debonding object
```
Comment (klmcadams) [inline examples/basic/fracture_analysis_contact_debonding.py:262]:
```suggestion
# Add contact debonding object
```

File: examples/basic/fracture_analysis_contact_debonding.py [hunk 7]
```diff
@@ -0,0 +1,405 @@
...
+
+MODEL.Activate()
+FRACTURE = MODEL.AddFracture()
+
+CONTACT_DEBONDING = FRACTURE.AddContactDebonding()
+CONTACT_DEBONDING.Material = MAT_CZM.Name
+CONTACT_DEBONDING.ContactRegion = CONTACT_REGION
+
+# %%
+# Define Analysis Settings
```
Comment (klmcadams) [inline examples/basic/fracture_analysis_contact_debonding.py:273]:
```suggestion
# Define analysis settings
```

File: examples/basic/fracture_analysis_contact_debonding.py [hunk 8]
```diff
@@ -0,0 +1,405 @@
...
+ani = FuncAnimation(
+    fig, update, frames=range(gif.n_frames), interval=100, repeat=True, blit=True
+)
+plt.show()
+
+
+app.save(os.path.join(cwd, "contact_debonding.mechdat"))
+app.new()
+
+# delete example file
```
Comment (klmcadams) [inline examples/basic/fracture_analysis_contact_debonding.py:404]:
```suggestion
# Delete example file
```

**Review by klmcadams — APPROVED** [review]

**Review by RobPasMue — APPROVED** [review]

File: examples/README.txt [hunk 9]
```diff
@@ -1,5 +1,5 @@
 Examples
 =========
 
-**Embedding examples** for `PyMechanical <_pymechanical_docs>`_ are contained in this repository.
+Embedding examples for `PyMechanical <_pymechanical_docs>`_ are contained in this repository.
```
Comment (sbradrick) [inline examples/README.txt:4]: Embedded, not embedding.

**Comment by dipinknair** [comment]
Thanks Sean,

I will make the changes in another PR, I will also look into all the examples for any documentation issues and will ask for your review


Thanks,

Dipin Nair /  R&D Engineer II

________________________________

**Pull request merged**